### PR TITLE
Add fortinet-fortigate package

### PR DESCRIPTION
**Pull request by luziegrutke**
This the first draft for the fortinet-fortigate package which I prepared together with Dominik. Examples are still missing.

### CODE REVIEW

File: fortinet-fortigate/package.yaml [hunk 1]
```diff
@@ -0,0 +1,38 @@
+id: fortinet.fortigate
```
Comment (dominiklohmann) [inline fortinet-fortigate/package.yaml:1]:
```suggestion
id: fortinet-fortigate
```

File: fortinet-fortigate/package.yaml [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+name: Fortinet FortiGate
+author: Tenzir
+author_icon: https://raw.githubusercontent.com/tenzir/library/main/author.svg
+package_icon: https://raw.githubusercontent.com/tenzir/library/main/fortinet-fortigate/package.svg
+
+description: |
+  [Fortinet FortiGate] (https://www.fortinet.com/products/next-generation-firewall) appliances provide network security by combining firewall, VPN, and intrusion prevention features.
```
Comment (dominiklohmann) [inline fortinet-fortigate/package.yaml:8]:
```suggestion
  [Fortinet FortiGate](https://www.fortinet.com/products/next-generation-firewall) appliances provide network security by combining firewall, VPN, and intrusion prevention features.
```

File: fortinet-fortigate/package.yaml [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+
+  This package makes it easy to onboard the logs that FortiGate generates.
```
Comment (dominiklohmann) [inline fortinet-fortigate/package.yaml:10]:
```suggestion
  This package makes it easy to onboard the logs that FortiGate generates, publishing the events to the `fortinet` topic.
```

File: fortinet-fortigate/package.yaml [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+      from "udp:// {{ inputs.syslog-ip-adress }}:{{ inputs.syslog-port }}" read syslog
+      | parse content kv
+      | yield content
+      | set #schema = "fortinet.fortigate"
+      | publish "fortinet"
+
+examples:
+
+       
+
```
Comment (dominiklohmann) [inline fortinet-fortigate/package.yaml:38]:
```suggestion
```

It's okay not to have examples for this one for now, I think. We need more experience with the data first.

File: fortinet-fortigate/package.yaml [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+    description: An UDP endpoint for receiving FortiGate logs via syslog.
+    type: string
+    default: 514
+
+pipelines:
+ onboard-from-udp:
+    name: Onboard FortiGate logs via UDP
+    description: A pipeline for ingesting and processing FortiGate logs received via UDP.
+    definition: |
+      from "udp:// {{ inputs.syslog-ip-adress }}:{{ inputs.syslog-port }}" read syslog
```
Comment (dominiklohmann) [inline fortinet-fortigate/package.yaml:29]:
```suggestion
      from "{{ inputs.syslog-url }}" read syslog
```

File: fortinet-fortigate/package.yaml [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+  syslog-ip-adress:
+    name: IP adress
+    description: A UDP endpoint where FortiGate sends its syslogs.
+    type: string
+    default: 127.0.0.1
+  syslog-port:
+    name: Syslog Port
+    description: An UDP endpoint for receiving FortiGate logs via syslog.
+    type: string
+    default: 514
```
Comment (dominiklohmann) [inline fortinet-fortigate/package.yaml:22]:
```suggestion
  syslog-url:
    name: Syslog URL
    description: The endpoint (including URL scheme) to which FortiGate sends syslog.
    type: string
    default: udp://0.0.0.0:514
```

File: fortinet-fortigate/package.yaml [hunk 7]
```diff
@@ -0,0 +1,38 @@
...
+
+pipelines:
+ onboard-from-udp:
```
Comment (dominiklohmann) [inline fortinet-fortigate/package.yaml:25]:
```suggestion
 onboard-fortigate:
```

**Review by dominiklohmann — APPROVED** [review]
This is a fantastic first PR—great to have you contribute to code now, too 🚀

File: fortinet-fortigate/package.yaml [hunk 8]
```diff
@@ -0,0 +1,28 @@
...
+pipelines:
+ onboard-fortigate:
+    name: Onboard FortiGate logs via UDP
+    description: A pipeline for ingesting and processing FortiGate logs received via UDP.
+    definition: |
+      from "{{ inputs.syslog-url }}" read syslog
+      | parse content kv
+      | yield content
+      | set #schema = "fortinet.fortigate"
+      | publish "fortinet"
```
Comment (dominiklohmann) [inline fortinet-fortigate/package.yaml:28]: Nitpicky comment: This file here has no newline at its end, as indicated by the icon here. Usually, it's considered good practice to end all files with a newlines. You can configure your text editor to automatically do that for you, too.

I'll take care of this now as part of signing the commits, but keep it in mind for the next one.

**Comment by mavam** [comment]
We should probably do a follow-up that TQL2'ifies the pipelines.

**Pull request merged**